extensions/cache/deployment/src/test/java/io/quarkus/cache/test/deployment/CacheConfigTest.java

```diff
@@ -29,11 +29,11 @@
     private static final String CACHE_NAME = "test-cache";
 
     @Inject
-    CacheRepository cacheRepository;
+    CacheManager cacheManager;
 
     @Test
     public void testConfig() {
-        CaffeineCache cache = (CaffeineCache) cacheRepository.getCache(CACHE_NAME);
+        CaffeineCache cache = cacheManager.getCache(CACHE_NAME).get().asCaffeineCache();
```

Comment:
Do we want asCaffeineCache? that is an impl detail and could need a `asOtherProvider26Cache` in the future
If you really want that feature, I'd use the unwrap strategy. `cacheManager.getCache(CACHE_NAME).get().asSpecializedCache(CaffeineCache.class)` //method name to be debated, it's usually called `unwrap` which can be more generic and thus better.

Reply:
Fixed.

Reply:
I took this out of the PR, #6903 will be fixed in a separate PR.